### PR TITLE
DNF normalizer

### PR DESCRIPTION
**Pull request by jakobbraun**
Added disjunctive normal form Normalizer in preparation for new selection predicate pushdown

### CODE REVIEW

File: src/main/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/DnfNormalizer.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * This class normalizes a {@link QueryPredicate} structure to the disjunctive normal form.
+ */
+@java.lang.SuppressWarnings("squid:S119") // DocumentVisitorType does not fit naming conventions.
+public class DnfNormalizer<DocumentVisitorType> {
+    private final QueryPredicateToLogicngConverter<DocumentVisitorType> converter;
+
+    /**
+     * Creates an instance of {@link DnfNormalizer}
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/DnfNormalizer.java:16]:
```suggestion
     * Create an instance of {@link DnfNormalizer}.
```

Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/DnfNormalizer.java:16]: I think we usually describe methods from `you` side: create, get, add etc. 
It not a mistake in your case, it's just nice to have them unified in our projects

File: src/main/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/QueryPredicateToLogicngConverter.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        predicate.accept(visitor);
+        return new Result<>(visitor.getFormula(), variablesMappingBuilder.getVariablesMapping());
+    }
+
+    private static class Visitor<DocumentVisitorType> implements QueryPredicateVisitor<DocumentVisitorType> {
+        private final VariablesMappingBuilder<DocumentVisitorType> variablesMappingBuilder;
+        private Formula formula;
+
+        private Visitor(final VariablesMappingBuilder<DocumentVisitorType> variablesMappingBuilder) {
+
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/QueryPredicateToLogicngConverter.java:38]: remove empty line

File: src/test/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/Selections.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.List;
+
+import com.exasol.adapter.dynamodb.documentnode.MockValueNode;
+import com.exasol.adapter.dynamodb.mapping.ToJsonPropertyToColumnMapping;
+import com.exasol.adapter.dynamodb.remotetablequery.ColumnLiteralComparisonPredicate;
+import com.exasol.adapter.dynamodb.remotetablequery.ComparisonPredicate;
+import com.exasol.adapter.dynamodb.remotetablequery.LogicalOperator;
+
+class Selections {
```
Comment (AnastasiiaSergienko) [inline src/test/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/Selections.java:11]: Can you add `constants` to the class name?

File: src/main/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/DnfNormalizer.java (hunk 1 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/adapter/dynamodb/remotetablequery/normalizer/DnfNormalizer.java]: ok. I tried to change it everywhere.

**Pull request merged**